libs/account-lib/src/types/dtos/accounts.request.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsNotEmpty } from 'class-validator';
+import { Signer, SignerPayloadRaw } from '@polkadot/types/types';
+import { TransactionType } from '#account-lib';
+
+export class RetireMsaRequestDto {
+  @ApiProperty({ type: Object })
+  @IsNotEmpty()
+  signerPayload: SignerPayloadRaw;
+
+  @ApiProperty({ type: String })
+  @IsNotEmpty()
+  encodedPayload: string;
+
+  @ApiProperty({ type: Object })
+  @IsNotEmpty()
+  signer: Signer;
```

Comment:
Suggestion: for `signerPayload` and `signer`, let's create a concrete `class` that implements each interface; this way we can actually use that type in the `@ApiProperty({ type: xxx })` decorator, and the OpenAPI spec will contain the correct structure for the payload, rather than just an opaque `Object`.

I think this is necessary for a useful public API spec.